stl/inc/vector

```diff
@@ -1208,7 +1208,7 @@ public:
                 _ASAN_VECTOR_MODIFY(static_cast<difference_type>(_Newsize - _Oldsize));
                 _Mylast = _Uninitialized_fill_n(_Mylast, _Newsize - _Oldsize, _Val, _Al);
             } else {
-                _ASAN_VECTOR_EXTEND_GUARD(static_cast<difference_type>(_Newsize - _Oldsize));
+                _ASAN_VECTOR_EXTEND_GUARD(_Newsize);

```

Comment:
All 48 runs of `GH_002030_asan_annotate_vector` pass for me if I revert this change, line 1293, and line 1951. I'd feel better if we add test coverage for all four of these holes and not just line 1409 that the Compiler team triggered.